Tests/Common/SimulatedDevice.cs

```diff
@@ -295,5 +294,16 @@ public void SetupJoin(AppSessionKey appSKey, NetworkSessionKey nwkSKey, DevAddr
             LoRaDevice.NwkSKey = nwkSKey;
             LoRaDevice.DevAddr = devAddr;
         }
+
+        private static byte[] StringToByteArray(string hex)
+        {
+            if (hex is null) throw new ArgumentNullException(nameof(hex));
+
+            var numberChars = hex.Length;
+            var bytes = new byte[numberChars / 2];
+            for (var i = 0; i < numberChars; i += 2)
```

Comment:
[nit] why not reuse `Hexadecimal` for this private method?

Reply:
Good point. Done with 4c4ed9f8e97b508237bebbe04081655e0ccb29ed.